contracts/portals/ENSPortal.sol

```diff
@@ -0,0 +1,57 @@
+// (c) Cartesi and individual authors (see AUTHORS)
+// SPDX-License-Identifier: Apache-2.0 (see LICENSE)
+
+pragma solidity ^0.8.8;
+
+import {ENS} from "@ensdomains/ens-contracts/contracts/registry/ENS.sol";
+
+import {IERC165} from "@openzeppelin/contracts/utils/introspection/IERC165.sol";
+
+import {IENSPortal} from "./IENSPortal.sol";
+import {Portal} from "./Portal.sol";
+import {IInputBox} from "../inputs/IInputBox.sol";
+import {InputEncoding} from "../common/InputEncoding.sol";
+import {LibENS} from "../library/LibENS.sol";
+
+/// @title ENS Portal
+///
+/// @notice This contract allows anyone to send their ENS to the InputBox
```

Comment:
If I understand you correctly, you are describing a scenario where someone will call this function regularly to send input blob. However, I think they should only call this function once to associate their address with the ENS node, and later on for the input blobs, call the regular `addInput` function. If there's any change to the ENS resolution, the next owner should call this function again. That would be more gas efficient, because we shouldn't be repeatedly doing ENS resolution when there's no change. Or am I missing something?

Reply:
The idea is that the user will not be identified in the back-end through their Ethereum address, but rather through their ENS name. This way, the back-end never has an outdated view of the ENS registry, because resolution will occur at runtime when someone executes an output that deals with ENS node hashes.